src/foam/nanos/crunch/ruler/rules.jrl

```diff
@@ -240,10 +240,6 @@ p({
   "operation":3,
   "after":false,
   "enabled":true,
-  predicate: {
-    class: 'foam.nanos.ruler.predicate.FScriptRulePredicate',
-    query: 'o !exists || o.status != n.status || o.isInRenewable != n.isInRenewable'
```

Comment:
does removing this entire predicate present issues for create?

is there any way to detect change between before and edited version?

should we provide a mechanism to indicate that the data have been edited by the user?

Reply:
Yes, believe this is an issue.  I recall that this is important.  @noodlemoodle do you remember?  Think the last work we did here was during Envoice Subscriptions.  
@AlexeyGreer please run the test suite.

Reply:
Review:
**InvokeEditBehaviourRuleAction** provides server side protection/enforcement of EditBehaviour.  
This rule is currently **disabled** (in the default crunch/ruler/rules.jrl). 
**SaveUCJDataOnGranted** must restrict when data is saved. Hitherto this was when something became GRANTED. 
SaveUCJDataOnGranted also checks for appropriate conditions in the rule, so the predicate could be removed allowing entry on no status change. (perhaps n.status == GRANTED).  Along with duplicating the check of InvokeEditBehaviourRuleAction which calls editBehaviour.maybeApplyEdit when status is GRANTED -> GRANTED and data has changed.

Reply:
Refactoring of SaveUCJDataOnGranted
before line 47, the IsInRenewable test, perform:
1. test for old status GRANTED, new status GRANTED
2. editBehaviour.maybeApplyEdit (copy from InvokeEditBehaviourRuleAction)
3. if true, allow save. 
Looks like isRenewable is also checking for old status GRANTED so may be able to combine.  IsRenewable is checking that data has not changed, and editBehaviour, of course is on data change.